plugin-core/src/main/java/appland/settings/AppMapApplicationSettings.java

```diff
@@ -57,6 +60,21 @@ public void setCliEnvironment(@NotNull Map<String, String> environment) {
         this.cliEnvironment = Map.copyOf(environment);
     }
 
+    /**
+     * Apply environment variables and API key to the command line.
+     */
+    public GeneralCommandLine applyServiceEnvironment(GeneralCommandLine commandLine) {
+        commandLine = commandLine.withParentEnvironmentType(cliPassParentEnv
+                ? GeneralCommandLine.ParentEnvironmentType.CONSOLE
+                : GeneralCommandLine.ParentEnvironmentType.NONE);
+        commandLine = commandLine.withEnvironment(cliEnvironment);
+        var key = apiKey;
+        if (isNotEmpty(key)) {
+            commandLine = commandLine.withEnvironment("APPMAP_API_KEY", key);
+        }
+        return commandLine;
+    }
```

Comment:
nit: Altough I don't like `final` on parameters or local variables for readability, reassigning parameters is often confusing and leads to bugs.

(I usually tag parameters with `@NotNull` as much as possible to help IntelliJ with its inspections.)

How about this instead?

```suggestion
public GeneralCommandLine applyServiceEnvironment(@NotNull GeneralCommandLine commandLine) {
        var key = apiKey;
        var environmentType = cliPassParentEnv
                ? GeneralCommandLine.ParentEnvironmentType.CONSOLE
                : GeneralCommandLine.ParentEnvironmentType.NONE;

        return commandLine.withParentEnvironmentType(environmentType)
                .withEnvironment(cliEnvironment)
                .withEnvironment(isNotEmpty(key) ? Map.of("APPMAP_API_KEY", key) : Map.of());
    }
```